src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -1626,23 +1634,72 @@ else if (ticket == TicketType.POST_TELEPORT) // post teleport
     }
 
     @ScarpetFunction(maxParams = -1)
-    public Value sample_noise(Context c, @Locator.Block BlockPos pos, String... noiseQueries) {
-        return Value.NULL;
-        /*
-        int mappedX = QuartPos.fromBlock(pos.getX());
-        int mappedY = QuartPos.fromBlock(pos.getY());
-        int mappedZ = QuartPos.fromBlock(pos.getZ());
-        Climate.Sampler mns = ((CarpetContext) c).s.getLevel().getChunkSource().getGenerator().climateSampler();
-        Map<Value, Value> ret = new HashMap<>();
-
-        if (noiseQueries.length == 0) {
-            noiseQueries = new String[]{"continentalness", "erosion", "weirdness", "temperature", "humidity", "depth"};
+    public Value compute_density_function(Context c, @Locator.Block BlockPos pos, String... densityFunctionQueries) {
+
+        Map<Value, Value> result = new HashMap<>();
+
+        ServerLevel level = ((CarpetContext) c).level();
+        Registry<DensityFunction> densityFunctionRegistry = level.registryAccess().registryOrThrow(Registries.DENSITY_FUNCTION);
+
+        if (densityFunctionQueries.length == 0) {
+            return ListValue.wrap(densityFunctionRegistry.keySet().stream().map(ResourceLocation::toString).map(StringValue::of));
```

Comment:
Didn't realize: Removes the `minecraft` namespace, like other functions do, and removes a map call
```suggestion
            return ListValue.wrap(densityFunctionRegistry.keySet().stream().map(ValueConversions::of));
```